PWGHF/TableProducer/HFTreeCreatorLbToLcPi.cxx

```diff
@@ -0,0 +1,366 @@
+                  collision::BCId,
+                  full::Pt,
+                  full::Eta,
+                  full::Phi,
+                  full::Y,
+                  full::MCflag);
+
+} // namespace o2::aod
+
+namespace o2::aod
+{
+namespace hf_track_index_alice3_pid
+{
+DECLARE_SOA_INDEX_COLUMN(Track, track); //!
+DECLARE_SOA_INDEX_COLUMN(RICH, rich);   //!
+DECLARE_SOA_INDEX_COLUMN(FRICH, frich); //!
+} // namespace hf_track_index_alice3_pid
+
+DECLARE_SOA_INDEX_TABLE_USER(HfTrackIndexALICE3PID, Tracks, "HFTRKIDXA3PID", //!
+                             hf_track_index_alice3_pid::TrackId,
+                             hf_track_index_alice3_pid::RICHId,
+                             hf_track_index_alice3_pid::FRICHId);
+} // namespace o2::aod
+
+struct Alice3PidIndexBuilder {
+  Builds<o2::aod::HfTrackIndexALICE3PID> index;
+  void init(o2::framework::InitContext&) {}
+};
+
+/// Writes the full information in an output TTree
+struct HfTreeCreatorLbToLcPi {
+  Produces<o2::aod::HfCandLbFull> rowCandidateFull;
+  Produces<o2::aod::HfCandLbFullEvents> rowCandidateFullEvents;
+  Produces<o2::aod::HfCandLbFullParticles> rowCandidateFullParticles;
+
+  void init(InitContext const&)
+  {
+  }
+
+  using TracksPID = soa::Join<aod::BigTracksPID, aod::HfTrackIndexALICE3PID>;
+  using ExtendedTracksPID = soa::Join<TracksPID, aod::TracksExtended>;
+
+  void process(aod::Collisions const& collisions,
+               aod::McCollisions const& mccollisions,
+               soa::Join<aod::HfCandLb, aod::HfCandLbMCRec, aod::HFSelLbToLcPiCandidate> const& candidates,
+               soa::Join<aod::HfCandProng3, aod::HfCandProng3MCRec, aod::HFSelLcCandidate> const& Lccandidates,
+               soa::Join<aod::McParticles, aod::HfCandLbMCGen> const& particles,
+               aod::BigTracksMC const& bigtracksmc,
+               ExtendedTracksPID const&,
+               aod::FRICHs const&,
+               aod::RICHs const&)
+  {
+
+    // Filling candidate properties
+    rowCandidateFull.reserve(candidates.size());
+    for (auto& candidate : candidates) {
+      auto fillTable = [&](int FunctionSelection,
+                           float FunctionInvMass,
+                           float FunctionCt,
+                           float FunctionY) {
+        if (FunctionSelection >= 1) {
+          auto LcCand = candidate.index0_as<soa::Join<aod::HfCandProng3, aod::HfCandProng3MCRec, aod::HFSelLcCandidate>>();
+          auto track0 = candidate.index1_as<ExtendedTracksPID>(); // daughter pion track
+          auto track1 = LcCand.index0_as<ExtendedTracksPID>();    // granddaughter tracks (lc decay particles)
+          auto track2 = LcCand.index1_as<ExtendedTracksPID>();
+          auto track3 = LcCand.index2_as<ExtendedTracksPID>();
+
+          auto RICHPi0 = -5000.0;
+          auto RICHTrk1Pi = -5000.0;
+          auto RICHTrk1p = -5000.0;
+          auto RICHTrk2K = -5000.0;
+          auto RICHTrk3Pi = -5000.0;
+          auto RICHTrk3p = -5000.0;
```

Comment:
Use consistent names
```suggestion
          auto RICHTrk0Pi = -5000.0;
          auto RICHTrk1Pi = -5000.0;
          auto RICHTrk1P = -5000.0;
          auto RICHTrk2K = -5000.0;
          auto RICHTrk3Pi = -5000.0;
          auto RICHTrk3P = -5000.0;
```

Reply:
You did not modify it everywhere. The current version will not compile.

Reply:
Fixed all labels, and checked that it will build and run